lib/ansible/module_utils/redfish_utils.py

```diff
@@ -0,0 +1,932 @@
+        data = response['data']
+        set_bios_attr_uri = data["@Redfish.Settings"]["SettingsObject"]["@odata.id"]
+
+        # Example: bios_attr = {\"name\":\"value\"}
+        bios_attr = "{\"" + attr['bios_attr_name'] + "\":\"" + attr['bios_attr_value'] + "\"}"
+        payload = {"Attributes": json.loads(bios_attr)}
+        response = self.patch_request(self.root_uri + set_bios_attr_uri, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        return {'ret': True}
+
+    def create_bios_config_job(self):
+        result = {}
+        key = "Bios"
+        jobs = "Jobs"
+
+        # Search for 'key' entry and extract URI from it
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        bios_uri = data[key]["@odata.id"]
+
+        # Extract proper URI
+        response = self.get_request(self.root_uri + bios_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        set_bios_attr_uri = data["@Redfish.Settings"]["SettingsObject"]["@odata.id"]
+
+        payload = {"TargetSettingsURI": set_bios_attr_uri, "RebootJobType": "PowerCycle"}
+        response = self.post_request(self.root_uri + self.manager_uri + "/" + jobs, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+
+        response_output = response['resp'].__dict__
+        job_id = response_output["headers"]["Location"]
+        job_id = re.search("JID_.+", job_id).group()
+        return {'ret': True, 'msg': 'Config job created', 'job_id': job_id}
+
+    def get_fan_inventory(self):
+        result = {}
+        fan_details = []
+        key = "Thermal"
+
+        # Go through list
+        for chassis_uri in self.chassis_uri_list:
+            response = self.get_request(self.root_uri + chassis_uri)
+            if response['ret'] is False:
+                return response
+            result['ret'] = True
+            data = response['data']
+            if key in data:
+                # match: found an entry for "Thermal" information = fans
+                thermal_uri = data[key]["@odata.id"]
+                response = self.get_request(self.root_uri + thermal_uri)
+                if response['ret'] is False:
+                    return response
+                result['ret'] = True
+                data = response['data']
+
+                for device in data[u'Fans']:
+                    fan = {}
+                    # There is more information available but this is most important
+                    fan['Name'] = device[u'FanName']
+                    fan['RPMs'] = device[u'Reading']
+                    fan['State'] = device[u'Status'][u'State']
+                    fan['Health'] = device[u'Status'][u'Health']
+                    fan_details.append(fan)
```

Comment:
Constructing dictionary key by key is not the most pythonic way of doing things. It would be way more common to have a single call in a loop like this:

    for device in data["Fans"]:
        fan_details.append(dict(
            Name=device["FanName"],
            RPMs=device["Reading"],
            ...
        ))

Reply:
I like this